File: java/Business/BusinessImpl/src/main/java/com/rallies/business/impl/TeamService.java
package com.rallies.business.impl;

import com.rallies.dataaccess.api.TeamRepository;
import com.rallies.domain.models.Team;

import java.util.Collection;
import java.util.Optional;

public class TeamService {
    private TeamRepository teamRepository;

    public TeamService(TeamRepository teamRepository) {
        this.teamRepository = teamRepository;
    }

    public Team addTeam(String teamName) {
        Team team = new Team(teamName);
        return teamRepository.save(team);
    }

    public Collection<Team> getAllTeams() {
        return teamRepository.getAll();
    }

    public Optional<Team> getTeamByName(String teamName) {
        return teamRepository.getTeamByName(teamName);
    }
}

File: csharp/Business/Services/Implementation/TeamService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


using Repository.Interfaces;
using Domain.Models;

namespace Business.Services
{
    public class TeamService
    {
        private ITeamRepository teamRepository;

        public TeamService(ITeamRepository teamRepository)
        {
            this.teamRepository = teamRepository;
        }

        public Team AddTeam(String teamName)
        {
            Team team = new Team(teamName);
            return teamRepository.Add(team);
        }

        public ICollection<Team> GetAllTeams()
        {
            return teamRepository.GetAll();
        }

        public Team? GetTeamByName(String teamName)
        {
            return teamRepository.GetByName(teamName); 
        }
    }
}
